pkg/daemon/update.go

```diff
@@ -100,7 +101,7 @@ func (dn *Daemon) updateOSAndReboot(newConfig *mcfgv1.MachineConfig) error {
 			return true, nil
 		})
 		if err != nil {
```

Comment:
I think we're really masking the error here from reporting if this times out, we should set err from line 91 to something like lastErr and check if we timed out here and report also that (otherwise `err == wait.ErrWaitTimeout` and that's gonna be it)

Reply:
Yeah...we need to wrap up that pattern in a helper function.

Reply:
Good catch!

Reply:
opened https://github.com/openshift/machine-config-operator/issues/410 to track this

Reply:
OK, fixed for now here too.

Reply:
I think I'm missing something here. If this times out, then we're propagating it up already, right? Note the `err` in the closure shadows the outer `err`, though I agree we should name them differently to make it more obvious.